src/plugins/intel_cpu/src/nodes/embedding_segments_sum.cpp

```diff
@@ -91,6 +91,12 @@ void EmbeddingSegmentsSum::initFromInputs() {
         numSegments_ = reinterpret_cast<const int *>(getParentEdgeAt(NUM_SEGMENTS_IDX)->getMemoryPtr()->GetPtr())[0];
     }
 
+    if (getChildEdgesAtPort(0)[0]->getMemoryPtr()->GetShape().isDynamic()) {
+        auto newDims = getParentEdgesAtPort(EMB_TABLE_IDX)[0]->getMemory().getStaticDims();
+        newDims[0] = numSegments_;
+        redefineOutputMemory({newDims});
+    }
```

Comment:
Another way is to redefineOutputMemory({newDims}) in [single place](https://github.com/openvinotoolkit/openvino/blob/d3d8fd8c202f408c7d816b5af85fe85cd0aa13c7/src/plugins/intel_cpu/src/node.cpp#L522).  In this implementation, we need override needShapeInfer() to cover dynamic shapes and dynamic values(save previous numSegments_), otherwise, it seems like called twice. Similar way in [interpolate](https://github.com/openvinotoolkit/openvino/blob/d3d8fd8c202f408c7d816b5af85fe85cd0aa13c7/src/plugins/intel_cpu/src/nodes/interpolate.cpp#L1782). I am not sure if this is better?

Reply:
Thanks @chenhu-wang, That's a good suggestion. Refactored accordingly.